# HG changeset patch
# User Philipp Weyer <pw@igh-essen.com>
# Date 1504538686 -7200
#      Mon Sep 04 17:24:46 2017 +0200
Fixed vm_fault.
(grafted from 8d6833867a56abc48afd5bfd645f14aacd8478fb)

diff --git a/master/cdev.c b/master/cdev.c
--- a/master/cdev.c
+++ b/master/cdev.c
@@ -62,7 +62,11 @@ static int eccdev_mmap(struct file *, st
 #define PAGE_FAULT_VERSION KERNEL_VERSION(2, 6, 23)
 
 #if LINUX_VERSION_CODE >= PAGE_FAULT_VERSION
-static int eccdev_vma_fault(struct vm_area_struct *, struct vm_fault *);
+static int eccdev_vma_fault(
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 11, 0)
+        struct vm_area_struct *,
+#endif
+        struct vm_fault *);
 #else
 static struct page *eccdev_vma_nopage(
         struct vm_area_struct *, unsigned long, int *);
@@ -278,8 +282,8 @@ static int eccdev_vma_fault(
     get_page(page);
     vmf->page = page;
 
-    EC_MASTER_DBG(priv->cdev->master, 1, "Vma fault, virtual_address = %p,"
-            " offset = %lu, page = %p\n", vmf->virtual_address, offset, page);
+    EC_MASTER_DBG(priv->cdev->master, 1, "Vma fault,"
+            " offset = %lu, page = %p\n", offset, page);
 
     return 0;
 }
